shopsafe-backend/src/main/java/com/google/sps/servlets/StoresServlet.java

```diff
@@ -283,4 +250,55 @@ private void addStore(
       return stores;
     }
   }
+
+  /** Returns a LatLng based on an address. */
+  public LatLng getLatLngFromAddress(String address) {
+
+    // Get a string array for all the words in the request and get its length.
+    String[] addressArray = address.trim().split("\\s+");
+    int addressWordCount = addressArray.length;
+
+    // Add all words to the string builder with '+' in between each word.
+    StringBuilder addressStringBuilder = new StringBuilder();
+    addressStringBuilder.append(addressArray[0]);
+    int index = 1;
+    while (index < addressWordCount) {
+      addressStringBuilder.append("+" + addressArray[index]);
+      index += 1;
+    }
+
+    // Define the address and initialize the location.
+    address = new String(addressStringBuilder);
+
+    // Get LatLng location based on address.
+    try {
+
+      // Read response of call to FCC API given lat and lng.
+      URL url = new URL(GEOCODE_URL + address + PLACE_KEY);
+      BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()));
+
+      // Store response in json, by reading each line.
+      StringBuilder json = new StringBuilder();
+      String line;
+      while ((line = reader.readLine()) != null) {
+        json.append(line);
+      }
+      reader.close();
+
+      // Convert json to json object with just the json location, then convert to LatLng.
+      JSONObject jsonLocation =
+          new JSONObject(new String(json))
+              .getJSONArray("results")
+              .getJSONObject(0)
+              .getJSONObject("geometry")
+              .getJSONObject("location");
+      return new LatLng(jsonLocation.getDouble("lat"), jsonLocation.getDouble("lng"));
+    }
+
+    // If error, print error, and return.
+    catch (Exception e) {
+      e.printStackTrace();
+      return new LatLng(1000, 1000);
+    }
+  }
```

Comment:
What do these changes do relative to the old version?

Reply:
Because I put it in a helper function, I need some way of handling the error case, so I assigned an invalid value that I check for. I dislike this solution myself and I would be open to a better idea if you have one!

Reply:
I'm an idiot, pointers would work so well here, plus I can just return true or false for failure. Thanks @caroljli

Reply:
Fixed it!